src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/DefaultRazorSemanticTokensInfoService.cs

```diff
@@ -109,12 +107,10 @@ internal class DefaultRazorSemanticTokensInfoService : RazorSemanticTokensInfoSe
             cancellationToken.ThrowIfCancellationRequested();
             var razorSemanticRanges = TagHelperSemanticRangeVisitor.VisitAllNodes(codeDocument, range);
             IReadOnlyList<SemanticRange>? csharpSemanticRanges = null;
-            string? newResultId = null;
-            var isFinalizedCSharp = false;
```

Comment:
I'm guessing based on this that the VSPlatform isn't "smart" about when it requests ranges but just tries again every Xms regardless? What happens if the C# request wasn't finalized and now we're returning the cached version?